File: solutions/python/build_name_by_number.py
'''
Created on Nov 15, 2019

@author: Ashish Namdev
'''
## Q1 Programming
### You are given  random build number, from the following data format **you need to provide respective Build name**
'''
Build Number Range - Build Name
000-010 - Hourly
011-020 - Nightly
021-030 - Beta
031-040 - Test
101-199 - Release Candidate
201-300 - Release
'''

VERSION = '1.0.0'
DATE = '2019-11-15'
UPDATED = '2019-11-15'

data = '''000-010 = Hourly
011-020 = Nightly
021-030 = Beta
031-040 = Test
101-199 = Release Candidate
201-300 = Release
'''

dict_data = {dt.split('=')[0].strip() : dt.split('=')[-1].strip() for dt in data.splitlines()}
build_no = 300
build_name = None

for key, val in dict_data.items():
  if build_no >= int(key.split('-')[0]) and build_no <= int(key.split('-')[-1]):
    build_name = val

print (build_name)

File: solutions/java/BuildNameByNumber.java
/*
Created on Nov 27, 2019

@author: Ashish Namdev
*/

/*
## Q1-Programming
### You are given  random build number, from the following data format **you need to provide respective Build name**
*/

import java.util.Map;
import java.util.LinkedHashMap;

public class BuildNameByNumber
{
    public static void main(final String []args)
    {
        int buildNumber = 251;
        String buildName = null;

        String data = "000-010 = Hourly\n";
        data += "011-020 = Nightly\n";
        data += "021-030 = Beta\n";
        data += "031-040 = Test\n";
        data += "101-199 = Release Candidate\n";
        data += "201-300 = Release\n";

        Map<String, String> dict = new LinkedHashMap<String, String>();
        for (String dt : data.split("\n"))
        {
            dict.put(dt.split("=")[0].trim(), dt.split("=")[1].trim());
        }

        for (Map.Entry<String, String> entry : dict.entrySet())
        {

            if(buildNumber >= Integer.parseInt(entry.getKey().split("-")[0])
                    && 
                buildNumber <= Integer.parseInt(entry.getKey().split("-")[1]))
            {
                buildName = entry.getValue();
                break;
            }
        }
        System.out.print(buildName);
    }
}